File: 782-jewels-and-stones/jewels-and-stones.cpp
class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        int cost = 0;
        map<char, int> stoneMap;
        for(auto stone : stones){
            stoneMap[stone]++;
        }
        for(auto jewel : jewels){
            cost += stoneMap[jewel];
        }
        return cost;
    }
};

File: 782-jewels-and-stones/jewels-and-stones.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        stoneMap = {}
        number = 0
        for stone in stones:
            if stone not in stoneMap:
                stoneMap[stone] = 1
            else:
                stoneMap[stone] += 1
        for jewel in jewels:
            if jewel in stoneMap:
                number += stoneMap[jewel]
        return number